metadata/metadata.go

```diff
@@ -164,6 +166,23 @@ func ExtractMeta(data []byte, spaceFromCli string, titleFromH1 bool, parents []s
 		meta.Parents = append(parents, meta.Parents...)
 	}
 
+	// deterministically generate a hash from the page's parents, space, and title
+	if titleAppendGeneratedHash {
+		log.Tracef(
```

Comment:
If you place the log below the hash generation, you can include the hash in the log info, also please use Debugf instead.